File: Python/maximum_subarray.py
def solve(nums):
    max_sum = nums[0]
    current_sum = 0
    for e in nums:
        if current_sum < 0:
            current_sum = 0
        current_sum += e
        max_sum = max(current_sum, max_sum)
    return max_sum


def main():
    nums = [-2, 1, -3, 4, -1, 2, 1, -5, 4]
    print(solve(nums))      # 6


if __name__ == '__main__':
    main()

File: Java/src/MaximumSubarray.java
public class MaximumSubarray {
    public static void main(String[] args) {
        int[] nums = {-2, 1, -3, 4, -1, 2, 1, -5, 4};
        System.out.println(solve(nums));    // 6
    }

    static int solve(int[] nums) {
        int maxSum = nums[0];
        int currentSum = 0;
        for (int e: nums) {
            if (currentSum < 0) currentSum = 0;
            currentSum += e;
            maxSum = Math.max(maxSum, currentSum);
        }
        return maxSum;
    }
}
